addons/sale/models/sale_order.py

```diff
@@ -520,6 +520,16 @@ def create(self, vals):
         result = super(SaleOrder, self).create(vals)
         return result
 
+    def write(self, vals):
+        if 'commitment_date' in vals:
+            if self.picking_ids:
+                for picking in self.picking_ids:
+                    if vals.get('commitment_date') is False:
+                        picking.write({"date_deadline": self.expected_date})
+                    else:
+                        picking.write({"date_deadline": vals.get('commitment_date')})
+        return super(SaleOrder, self).write(vals)
+
```

Comment:
Hi
- You should loop in self (you can have multi order in self)
- `return super(SaleOrder, self).write(vals)` --> `return super().write(vals)`
- filtered pickings (only take no done and no cancelled)
- use sudo, if saleman have no access to stock

```python
    def write(self, vals):
        if 'commitment_date' in vals:
            for order in self.sudo():
                  order.picking_ids.filtered(lambda x: x.state not in ('done','cancel')).write({'date_deadline': vals['commitment_date'] or order.expected_date})
        return super().write(vals)
```